http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -145,7 +158,14 @@ private JettyHttpClient(HttpClientConfig config, Optional<JettyIoPool> jettyIoPo
             sslContextFactory.setKeyStorePassword(config.getKeyStorePassword());
         }
 
-        httpClient = new HttpClient(new HttpClientTransportOverHTTP(2), sslContextFactory);
+        if (authenticationEnabled) {
+            requireNonNull(kerberosConfig.getConfig(), "kerberos config path is null");
+            requireNonNull(config.getKerberosRemoteServiceName(), "kerberos remote service name is null");
+            httpClient = new WrappedHttpClient(kerberosConfig, config, new HttpClientTransportOverHTTP(2), sslContextFactory);
+        }
+        else {
+            httpClient = new HttpClient(new HttpClientTransportOverHTTP(2), sslContextFactory);
```

Comment:
maybe declare `new HttpClientTransportOverHTTP(2)` outside of the `if` statement